csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go

```diff
@@ -424,6 +428,28 @@ func (p *csiProvisioner) Provision(options controller.ProvisionOptions) (*v1.Per
 	return pv, err
 }
 
+type CustomPropertiesSpec map[string]interface{}
+
+func (cps CustomPropertiesSpec) IsEmpty() bool {
+	if nil == cps {
+		return true
+	}
+	return false
+}
+
+func (cps CustomPropertiesSpec) GetDriverPreference() string {
+	var driverName string
+	if cps.IsEmpty() {
+		return "NoDriverFound"
+	}
+	for k, v := range cps {
+		if k == "driver" {
+			driverName = fmt.Sprintf("%v", v)
+			fmt.Println(driverName)
```

Comment:
Can we use logger? And also log proper error message

Reply:
removed the fmt code, this was just for development debugging